### PR TITLE
Merge deepmd-pytorch into main repo

### PR DESCRIPTION
**Pull request by iProzd**
Merge the deepmd-pytorch into main repo🎉
Add the following directories:

- deepmd/pt : main implementations of deepmd-pytorch
- source/tests/pt: UTs for deepmd-pytorch

TODO list:
- [x] examples added for water/se_e2_a, water/se_atten, water/dpa2
- [x] README updated (need modified)
- [x] Paths in each files have been adapted. 
- [x] pyproject.toml needed to be merge

### CODE REVIEW

**Comment by njzjz** [comment]
I give up testing TF 2.2. It requires NumPy < 1.20, while the latest h5py requires NumPy >= 1.22... It takes a lot of work to pin these requirements.
I think it's okay, as `tf.compat.v1` has been stable since v2...

**Comment by wanghan-iapcm** [comment]
> I give up testing TF 2.2. It requires NumPy < 1.20, while the latest h5py requires NumPy >= 1.22... It takes a lot of work to pin these requirements. I think it's okay, as `tf.compat.v1` has been stable since v2...

I agree

File: deepmd/pt/model/descriptor/repformers.py [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+        """
+        super().__init__()
+        del type
+        self.epsilon = 1e-4  # protection of 1./nnei
+        self.rcut = rcut
+        self.rcut_smth = rcut_smth
+        self.ntypes = ntypes
+        self.nlayers = nlayers
+        sel = [sel] if isinstance(sel, int) else sel
+        self.nnei = sum(sel)  # 总的邻居数量
```
Comment (njzjz) [inline deepmd/pt/model/descriptor/repformers.py:100]: Chinese should be avoided in the comment...

**Review by njzjz — APPROVED** [review]
All tests have passed 🎉

**Pull request merged**